packages/compose/src/hooks/use-shallow-compare-effect/index.js

```diff
@@ -0,0 +1,26 @@
+/**
+ * WordPress dependencies
+ */
+import { useEffect, useRef } from '@wordpress/element';
+import isShallowEqual from '@wordpress/is-shallow-equal';
+
+/**
+ * Like `useEffect` but call the effect when the dependencies are not shallowly equal.
+ * Useful when the size of the dependency array might change during re-renders.
+ * This hook is only used for backward-compatibility reason. Consider using `useEffect` wherever possible.
+ *
+ * @param {Function} effect The effect callback passed to `useEffect`.
+ * @param {Array}    deps   The dependency array that is compared against shallowly.
+ */
+function useShallowCompareEffect( effect, deps ) {
```

Comment:
Why do we need new APIs just to work around the rule? Why can't we disable the rule for this particular instance?

I don't understand the need to introduce new APIs at all in this PR, unless it's going to be used in multiple places. Maybe we should just focus on fixing the dependencies in this PR?

Reply:
Hmm, yeah it makes sense, not sure I included it in the first place.